libraries/AP_Proximity/AP_Proximity_LD06.cpp

```diff
@@ -0,0 +1,172 @@
+                parse_response_data();
+                _temp_boundary.update_3D_boundary(boundary);
+                _temp_boundary.reset();
+                
+
+                // Resets the bytes read and whether or not we are reading data to accept a new payload
+                _byte_count = 0;
+                _response_data = false;
+            }
+        }
+    }
+}
+
+// Parses the data packet received from the LiDAR
+void AP_Proximity_LD06::parse_response_data()
+{ 
+
+    // Data interpretation based on:
+    // http://wiki.inno-maker.com/display/HOMEPAGE/LD06?preview=/6949506/6949511/LDROBOT_LD06_Development%20manual_v1.0_en.pdf
+
+    // Second byte in array stores length of data - not used but stored for debugging
+    // const uint8_t data_length = _response[START_DATA_LENGTH];
+
+    // Respective bits store the radar speed, start/end angles
+    // Use bitwise operations to correctly obtain correct angles
+    // Divide angles by 100 as per manual
+    const float start_angle = float(UINT16_VALUE(_response[START_BEGIN_ANGLE + 1], _response[START_BEGIN_ANGLE])) * 0.01;
+    const float end_angle = float(UINT16_VALUE(_response[START_END_ANGLE + 1], _response[START_END_ANGLE])) * 0.01;
+
+    // Verify the checksum that is stored in the last element of the response array
+    // Return if checksum is incorrect - i.e. bad data, bad readings, etc.
+    const uint8_t check_sum = _response[START_CHECK_SUM];
+    if (check_sum != crc8_generic(&_response[0], sizeof(_response) / sizeof(_response[0]) - 1, 0x4D)) {
+        return;
+    }
+
+    // Calculates the angle that this point was sampled at
+    float sampled_counts = 0;
+    const float angle_step = (end_angle - start_angle) /  (PAYLOAD_COUNT - 1);
+    float uncorrected_angle = start_angle + (end_angle - start_angle) * 0.5;
+
+    // Handles the case that the angles read went from 360 to 0 (jumped)
+    if (angle_step < 0) {
+        uncorrected_angle = wrap_360(start_angle + (end_angle + 360 - start_angle) * 0.5);
+    }
+
+    // Takes the angle in the middle of the readings to be pushed to the database
+    const float yaw_correction = frontend.get_yaw_correction(state.instance);
+    const float push_angle = (frontend.get_orientation(state.instance) == 1) ? 
+        wrap_360(180 + (uncorrected_angle + yaw_correction)) : 
+        wrap_360(uncorrected_angle + yaw_correction);
+    
+    float distance_avg = 0.0;
+
+    // Each recording point is three bytes long, goes through all of that and updates database
+    for (uint16_t i = START_PAYLOAD; i < START_PAYLOAD + MEASUREMENT_PAYLOAD_LENGTH * PAYLOAD_COUNT; i += MEASUREMENT_PAYLOAD_LENGTH) {
+
+        // Gets the distance recorded and converts to meters
+        const float distance_meas = UINT16_VALUE(_response[i + 1], _response[i]) * 0.001;
+
+        // Validates data and checks if it should be included
+        if (distance_meas > distance_min() && distance_meas < distance_max()) {
```

Comment:
what does this sensor do if it has no targets in view?

Reply:
If there are no targets in view (or the targets are outside of the boundaries that the sensor can reliably read input at), the distance returned will be 0, and that will lead to the "sampled_counts" variable not registering enough inputs to push a reading to the Proximity Boundary. The face will just be empty then, as it should be if there is no input returned.